akka-actor-tests/src/test/java/akka/actor/JavaExtension.java

```diff
@@ -34,7 +34,11 @@ public TestExtension(ExtendedActorSystem i) {
   }
 
   static class OtherExtension implements Extension {
-    static final ExtensionKey<OtherExtension> key = new ExtensionKey<OtherExtension>(OtherExtension.class) {
+    static final ExtensionId<OtherExtension> key = new AbstractExtensionId<OtherExtension>() {
```

Comment:
I actually don't understand quite what the `OtherExtension` is testing/showing an example of here. Is that because it got superfluous when `ExtensionKey` was removed? The test that uses it is called `mustBeAdHoc`